### PR TITLE
Moment removal

### PR DESCRIPTION
**Pull request by Darrken**
This PR removes moment as a dependency and uses the native javascript `Date` object paired with date-fns for all date functionality within the datepicker.
Addresses #1349 (among other similar requests)

Benefits:
- much smaller dependency footprint
- no more clone() everywhere; date-fns functions return a new date object
- downstream code doesn't need any specific library to handle dates

Removed:
- moment.js
- `utcOffset` as a prop - if anyone has any suggestions how this might be implemented with the native `Date` object, I'd love to hear ideas. Otherwise, I felt leaving this out was likely an acceptable trade-off for now. (date-fns does have a UTC module in development.)
- Timezone date demo

Added:
- some public helper functions to make up for the localization support moment provided: `registerLocale, setDefaultLocale, getDefaultLocale` (docs updated to reflect these changes)

This is obviously a pretty big PR, so thorough review is needed and welcome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,23 +20,22 @@ The package can be installed via NPM:
 npm install react-datepicker --save
 ```
 
-You’ll need to install React, PropTypes, and Moment.js separately since those dependencies aren’t included in the package. Below is a simple example of how to use the Datepicker in a React view. You will also need to require the CSS file from this package (or provide your own). The example below shows how to include the CSS from this package if your build system supports requiring CSS files (Webpack is one that does).
+You’ll need to install React, PropTypes, and date-fns (for localization support if needed) separately since those dependencies aren’t included in the package. Below is a simple example of how to use the Datepicker in a React view. You will also need to require the CSS file from this package (or provide your own). The example below shows how to include the CSS from this package if your build system supports requiring CSS files (Webpack is one that does).
```
Comment (aabenoja) [inline README.md]: I'm slightly confused by the start of this line. We have `date-fns` set as a dependency, but we're telling users to install it like we would a peer dependency?

Comment (Darrken) [reply README.md]: You're right, I was thinking they'd need it in their downstream project but they shouldn't. I'll make that more clear.

**Comment by jRichardeau** [comment]
Do you know when this PR could be available ? I want to use this picker but moment is a too big dependency.
Thank you

**Comment by Darrken** [comment]
@aij Date_utils still serves as an abstraction, at least the same way it did for all the moment functions.  I don't think I could have done this without the previous work to move the moment logic into date_utils! I tried to keep any calls to `new Date()` or date-fns functions behind the date_utils functions so there's a single source. If you have a suggestion on how that could better be separated or abstracted (maybe I'm misunderstanding), let me know.

**Comment by 4ware** [comment]
Please merge this into the master if no conflicts exist!!

**Comment by martijnrusschen** [comment]
Thanks, I merged it. Going to release a new major version today to reflect the breaking changes.

**Comment by martijnrusschen** [comment]
V2.0.0 is now live!

**Pull request merged**